lib/vrl/parser/src/parser.lalrpop

```diff
@@ -242,15 +244,22 @@ Assignment: Node<Assignment> = {
     Sp<AssignmentInfallible>,
 };
 
+AssignmentOp: AssignmentOp = {
+    "=" => AssignmentOp::Assign,
+    "|=" => AssignmentOp::Merge,
+}
+
 AssignmentSingle: Assignment = {
-    <target: Sp<AssignmentTarget>> "="
-        <expr: Box<Expr>> => Assignment::Single { target, expr },
+    <target: Sp<AssignmentTarget>>
+        <op: AssignmentOp>
```

Comment:
One more note (again, nothing you need to fix in this PR):

In general, I've tried to get as much span information from the parser as possible, so that our AST consistently provides span details for anything that's relevant for consumers of the AST.

In this case, the `=` token wasn't relevant before because there was no other alternative to use. Now that it can be either `=` or `|=`, it makes sense to include span information for the assignment operator as well.

We can tackle this in a follow-up PR after 0.12 ships though, as we don't actually need that span right now.